From 6e283a898c2271759daabd6873d4ee18a1e9b9ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Steven!=20Ragnar=C3=B6k?= <steven@nuclearsandwich.com>
Date: Thu, 11 Mar 2021 15:42:22 -0800
Subject: [PATCH 1/3] Source sh-compatible rtisetenv script before building.

In order for this package to properly detect the installed Connext DDS
installation the setup script must be sourced before build actions.
Since the build runs in a sh-compatible setting the bash-specific
functionality of the script must be patched out.
---
 debian/rules.em | 22 +++++++++++++++++-----
 1 file changed, 17 insertions(+), 5 deletions(-)

diff --git a/debian/rules.em b/debian/rules.em
index a856369..0e16329 100755
--- a/debian/rules.em
+++ b/debian/rules.em
@@ -26,10 +26,11 @@ DEB_HOST_GNU_TYPE ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
 %:
 	dh $@@ -v --buildsystem=cmake --builddirectory=.obj-$(DEB_HOST_GNU_TYPE)
 
-override_dh_auto_configure:
+override_dh_auto_configure: /tmp/rtienv.sh
 	# In case we're installing to a non-standard location, look for a setup.sh
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+	. /tmp/rtienv.sh && \
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
 	dh_auto_configure -- \
 		-DCMAKE_INSTALL_PREFIX="@(InstallationPrefix)" \
@@ -37,31 +38,42 @@ override_dh_auto_configure:
 		-DCMAKE_PREFIX_PATH="@(InstallationPrefix)" \
 		$(BUILD_TESTING_ARG)
 
-override_dh_auto_build:
+override_dh_auto_build: /tmp/rtienv.sh
 	# In case we're installing to a non-standard location, look for a setup.sh
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+	. /tmp/rtienv.sh && \
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
 	dh_auto_build
 
-override_dh_auto_test:
+override_dh_auto_test: /tmp/rtienv.sh
 	# In case we're installing to a non-standard location, look for a setup.sh
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+	. /tmp/rtienv.sh && \
 	echo -- Running tests. Even if one of them fails the build is not canceled.
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
 	dh_auto_test || true
 
-override_dh_shlibdeps:
+override_dh_shlibdeps: /tmp/rtienv.sh
 	# In case we're installing to a non-standard location, look for a setup.sh
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+	. /tmp/rtienv.sh && \
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
 	dh_shlibdeps -l$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/lib/:$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/opt/@(Name)/lib/
 
-override_dh_auto_install:
+override_dh_auto_install: /tmp/rtienv.sh
 	# In case we're installing to a non-standard location, look for a setup.sh
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
+	. /tmp/rtienv.sh && \
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
 	dh_auto_install
+
+/tmp/rtienv.sh:
+	if [ -f /opt/rti.com/rti_connext_dds-5.3.1/resource/scripts/rtisetenv_x64Linux3gcc5.4.0.bash ]; then \
+	sed -e 's:\$${BASH_SOURCE\[0\]}:/opt/rti.com/rti_connext_dds-5.3.1/resource/scripts/rtisetenv_x64Linux3gcc5.4.0.bash:' \
+	/opt/rti.com/rti_connext_dds-5.3.1/resource/scripts/rtisetenv_x64Linux3gcc5.4.0.bash \
+	> /tmp/rtienv.sh; \
+	else touch /tmp/rtienv.sh; fi
-- 
2.43.0

